### PR TITLE
Extended StepActions functionality

### PR DESCRIPTION
**Pull request by tsnowlan**
* Adds helper functions for escalation
* StepActions now support all StepAction options: `complete`, `completerepeat`, `nextstep`, `remove`, `repeat`, `review` and `rework`
* `Step.create` now functions as desired
* DateTimeDescriptor class 
* Misc whitespace cleanup

### CODE REVIEW

**Comment by tsnowlan** [comment]
This should be good to go, but could use some extra eyes to see that I didn't miss anything.

**Comment by handstad** [comment]
I don't think I have enough background knowledge to really understand this in depth, but as far I can tell this looks good to me. Seems like we now have the ability to move samples however we want, that is good. I see that SciLifeLab has limited commenting in their code, I hope things are understandable if you have the necessary background. I have to leave the rest of the review to @xuyangy

**Comment by xuyangy** [comment]
@handstad @sveint Can I have  write access to merge this PR?

**Comment by handstad** [comment]
You should have write access now.

File: genologics/entities.py [hunk 1]
```diff
@@ -949,12 +993,30 @@ def get_next_actions(self):
...
-            action = [action for action in actions if action['artifact'].uri == art_uri][0]
-            if 'action' in action: node.attrib['action'] = action.get('action')
+            action_opts = [action for action in actions if action['artifact'].uri == art_uri][0]
+            if 'action' not in action_opts:
+                raise RuntimeError("No action type specified: {}".format(action_opts))
+            elif action_opts['action'] not in self._action_attrs:
+                raise RuntimeError("Invalid action type specified: {}".format(action_opts))
+
+            node.attrib['action'] = action_opts['action']
+            if action_opts['action'] == 'nextstep':
```
Comment (xuyangy) [inline genologics/entities.py]: Should this be removed so all types of actions can be set? However, 'review' may need also to set escalation?

**Review by xuyangy — APPROVED** [review]

File: genologics/entities.py (hunk 1 above)
Comment (tsnowlan) [reply genologics/entities.py]: ah, yeah, that can be pulled out. good catch, I'll update it

**Review by xuyangy — APPROVED** [review]

**Pull request merged**